src/main/java/org/takes/rs/RsWithHeader.java

```diff
@@ -94,22 +95,61 @@ public RsWithHeader(final Response res, final CharSequence name,
      */
     public RsWithHeader(final Response res, final CharSequence header) {
         super(
-            new Response() {
-                @Override
-                public List<String> head() throws IOException {
-                    final List<String> head = new LinkedList<String>();
-                    for (final String hdr : res.head()) {
-                        head.add(hdr);
-                    }
-                    head.add(header.toString());
-                    return head;
-                }
-                @Override
-                public InputStream body() throws IOException {
-                    return res.body();
-                }
-            }
+            new RsWithHeader.ResponseImpl(res, header)
+        );
+    }
+
+    private static final class ResponseImpl implements Response {
```

Comment:
@YamStranger please change class name according to http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#class-names